### PR TITLE
Feature/com 165 lap system

### PR DESCRIPTION
**Pull request by Lukauigi**

### CODE REVIEW

File: Game/include/Scripts/LapCheckpointScript.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#include "LapManagerScript.h"
+#include "../../../Physics/include/Aabb.h"
+
+class LapCheckpointScript : public Script
+{
+public:
+	LapCheckpointScript(Entity* mEntity);
+	~LapCheckpointScript();
+
+	void Init(LapManagerScript* lapManager, TransformComponent* other, int checkpointIndex);
```
Comment (Tissure) [inline Game/include/Scripts/LapCheckpointScript.h:13]: Change and override to SetParameters().
Conform to Script.h format and extend ScriptParameters struct.
Use OrbitScript for reference.

File: Game/include/Scripts/LapCheckpointScript.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	void SetCheckpointRegistered(bool isRegistered) { m_Registered = isRegistered; }
+
+	void Update(float deltaTime) override;
+	std::unique_ptr<Script> clone() const override {
+		return std::make_unique<LapCheckpointScript>(*this);
+	}
+private:
+	TransformComponent* m_Self = nullptr;
+	TransformComponent* m_Other = nullptr;
+	AE86::Aabb* m_SelfBB = nullptr;
```
Comment (Tissure) [inline Game/include/Scripts/LapCheckpointScript.h:24]: I don't know if you will need these AABB objects. Talk to physics for collision detection.

File: Game/include/Scripts/LapManagerScript.h
Comment (Tissure) [inline Game/include/Scripts/LapManagerScript.h:1]: Conform to Script.h format and extend ScriptParameters struct.
Use OrbitScript for reference.

**Review by Tissure — CHANGES_REQUESTED** [review]
Scripting changes from me. Gonna let physics handle the AABB.
Also check for merge conflicts before making a PR.

**Review by Tissure — APPROVED** [review]

**Review by imnorookie — APPROVED** [review]
lgtm :3 thanks 4 the hard work!

**Pull request merged**